ecl/eclagent/eclagent.cpp

```diff
@@ -2528,6 +2536,13 @@ void EclAgentWorkflowMachine::noteTiming(unsigned wfid, timestamp_type startTime
     const cost_type diskAccessCost = aggregateDiskAccessCost(wu, scope);
     if (diskAccessCost)
         wu->setStatistic(queryStatisticsComponentType(), queryStatisticsComponentName(), SSTworkflow, scope, StCostFileAccess, NULL, diskAccessCost, 1, 0, StatsMergeReplace);
+    stat_type totalSizeSpill = 0, peakSizeSpill = 0;
```

Comment:
Are these timings recorded if the workunit fails (e.g. because out of spill file)?  I suspect not.

Reply:
I don't think they are.  Can this be done in a separate Jira?

Reply:
it looks like the way exceptions can be thrown in perform() means noteTiming will be skipped - therefore the workflow item's noteTiming values will be lost.
But the total aggregate (- the missing failing workflow item ) will be recorded afaics.

NB: it's not just these new spill sizes which will be lost, it's also Started,  Elapsed, Cost .. that won't be recorded for the failing workflow item afaics.

Reply:
@shamser - I think a JIRA was opened for this(?) Could you link it into HPCC-28756?